src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -1305,6 +1305,28 @@ public GraphNode GetFirstDirtyGraphNode(int pc, int classIndex, int procIndex)
             return null;
         }
 
+        /// <summary>
+        /// Gets the first dirty graphnode
+        /// </summary>
+        /// <param name="pc"></param>
+        /// <param name="classIndex"></param>
+        /// <param name="procIndex"></param>
+        /// <returns></returns>
+        public GraphNode GetFirstDirtyGraphNode(int classIndex, int procIndex)
```

Comment:
We already has a nearly same function ``GetFirstDirtyGraphNode(int pc, int classIndex, int procIndex)`` and they have similar implementation. I'd suggest to combine these two together.